hack/windows/Install-OVS.ps1

```diff
@@ -30,64 +25,39 @@
   Specifies the path of a local SSL package to be used for installation.
 #>
 Param(
-    [parameter(Mandatory = $false)] [string] $DownloadDir,
     [parameter(Mandatory = $false)] [string] $DownloadURL,
     [parameter(Mandatory = $false)] [string] $OVSInstallDir = "C:\openvswitch",
-    [parameter(Mandatory = $false)] [bool] $CheckFileHash = $true,
     [parameter(Mandatory = $false)] [string] $LocalFile,
-    [parameter(Mandatory = $false)] [bool] $ImportCertificate = $true,
     [parameter(Mandatory = $false)] [bool] $InstallUserspace = $true,
     [parameter(Mandatory = $false)] [string] $LocalSSLFile
 )
 
 $ErrorActionPreference = "Stop"
-$OVSDownloadURL = "https://downloads.antrea.io/ovs/ovs-3.0.5-antrea.1-win64.zip"
+$DefaultOVSDownloadURL = "https://downloads.antrea.io/ovs/ovs-3.0.5-antrea.1-win64.zip"
 # Use a SHA256 hash to ensure that the downloaded archive is correct.
-$OVSPublishedHash = '813a0c32067f40ce4aca9ceb7cd745a120e26906e9266d13cc8bf75b147bb6a5'
+$DefaultOVSPublishedHash = '813a0c32067f40ce4aca9ceb7cd745a120e26906e9266d13cc8bf75b147bb6a5'
+# $MininalVCRedistVersion is the minimal version required by the provided Windows OVS binary. If a higher
+# version of VC redistributable file exists on the Windows host, we can skip the installation.
+$MininalVCRedistVersion="14.12.25810"
+$DefaultVCRedistsDownloadURL = "https://aka.ms/vs/17/release/vc_redist.x64.exe"
+
+$invocationName=$($myInvocation.MyCommand.Name)
 $WorkDir = [System.IO.Path]::GetDirectoryName($myInvocation.MyCommand.Definition)
-$OVSDownloadDir = $WorkDir
+$InstallLog = "$WorkDir\install_ovs.log"
 $PowerShellModuleBase = "C:\Windows\System32\WindowsPowerShell\v1.0\Modules"
-
-if (!$LocalFile) {
-    $OVSZip = "$OVSDownloadDir\ovs-win64.zip"
-} else {
-    $OVSZip = $LocalFile
-    $DownloadDir = Split-Path -Path $LocalFile
-}
-
-if ($DownloadDir -ne "") {
-    $OVSDownloadDir = $DownloadDir
-}
-
-$InstallLog = "$OVSDownloadDir\install_ovs.log"
-
-if ($DownloadURL -ne "") {
-    $OVSDownloadURL = $DownloadURL
-    # For user-provided URLs, do not verify the hash for the archive.
-    $OVSPublishedHash = ""
-}
+$OVSZip=""
+$DefaultOVSRunDir = "C:\openvswitch\var\run\openvswitch"
+# $global:updateDriver is a flag used to indicate whether the OVSext driver needs to be updated on the target
+# Windows host. We also use it as a flag to copy directory "openvswitch/usr/bin" from container to host path
+# c:/openvswitch/, in which the OVS utilities are provided, e.g., ovs-vsctl.exe, ovs-ofctl.exe, ovs-appctl.exe.
+# These OVS utilities are needed for local troubleshooting.
```

Comment:
nit: Not just for troubleshooting actually. For example, I think the antrea-agent relies on `ovs-appctl` to check supported DP features, even on Windows?

Reply:
Yes, you are right. I don't have a good idea about how to refine the descriptions here, one thing I want to explain is, 

1. If we run the installation script with option `$InstallUserspace=$true`, all sub-directories of openvswitch are copied to the install path, in this case the directory of `user/bin` is included, so we don't need an additional copy; 
2. If we want to run OVS inside the container, the OVS utility binaries are included in the container image, so antrea-agent can access the bits from container paths (we also add the paths into antrea-agent container's environment path) but not need a copy on the host. In this case, we do this copy is for the scenario that login on the Windows host with a local trouble shooting but not from antrea-agent container.

I use "local troubleshooting" is for case 2, but maybe my descriptions are not clear.

Reply:
I think the name of the flag makes it quite confusing.
Case 1 is taken care of by your code below:
```
    if ($InstallUserspace -eq $true) {
        InstallOVSServices($OVSInstallPath)
    } else {
        # Copy OVS utilities from container path "openvswitch/usr/bin" to host path "c:/openvswitch/"
        CopyOVSUtilities($OVSInstallPath)
    }
```

For case 2, could we just remove this flag and unconditionally copy the binaries. It feels like we could simplify the `CopyOVSUtilities` as follows:
```
function CopyOVSUtilities() {
    param (
        [parameter(Mandatory = $true)] [string] $OVSInstallPath
    )
    $hostUsrBinPath = "C:\openvswitch\usr\bin"
    $usrBinPath="${OVSInstallPath}\usr\bin"
    # I assume that this is only true in the "containerized" case?
    if ("$usrBinPath" -ne "$hostUsrBinPath") {
        mkdir -p $hostUsrBinPath
        cp -r $usrBinPath\* $hostUsrBinPath\
    }
    AddToEnvPath($hostUsrBinPath)
}
```
We just copy the files unconditionally. It doesn't matter if they already exist or not. This is similar to how we would install CNI files for Linux (https://github.com/antrea-io/antrea/blob/main/build/images/scripts/install_cni). What do you think?

Reply:
>  I assume that this is only true in the "containerized" case?

Yes, the case `"$usrBinPath" -ne "$hostUsrBinPath"` in function "CopyOVSUtilities" is true only in the containerized case.

I don't want to copy the bits every time in the driver-installation, because I feel it is not necessary to do so. A major difference from the OVS installation on Linux or CNI installation is, the number of copied files are small in those two cases, but the Windows OVS bin files are more (about over 20 files in the directory).  I would prefer to keep the flag `$global:updateDriver` which is set as true when OVSext driver is updated to re-install the OVS bits, so I want to update the function like this, 

```
function CopyOVSUtilities() {
    param (
        [parameter(Mandatory = $true)] [string] $OVSInstallPath
    )
    $hostUsrBinPath = "C:\openvswitch\usr\bin"
    $usrBinPath="${OVSInstallPath}\usr\bin"
    if ("$usrBinPath" -ne "$hostUsrBinPath") {
        if (-not $global:updateDriver) {
            return
        }
        mkdir -p $hostUsrBinPath
        cp -r $usrBinPath\* $hostUsrBinPath
        AddToEnvPath($hostUsrBinPath)
    }
}
```

If you think the flag `$global:updateDriver` is confusing, what about renaming it as "$global:updateOVS" ?